src/lapi.js

```diff
@@ -428,6 +428,14 @@ const lua_createtable = function(L, narray, nrec) {
     assert(L.top <= L.ci.top, "stack overflow");
 };
 
+const lua_newuserdata = function(L, u) {
+    L.stack[L.top++] = new lobject.TValue(CT.LUA_TUSERDATA, u);
```

Comment:
newuserdata takes a size. lobject.TValue takes a value.

You should probably create an `ArrayBuffer`.